### PR TITLE
[DSVA-15904] - ViT for YoloX

### PR DESCRIPTION
**Pull request by Pawel842**

### CODE REVIEW

File: yolox/models/network_blocks.py [hunk 1]
```diff
@@ -345,3 +345,93 @@ def forward(self, x):
...
+        self.q = nn.Linear(c, c, bias=False)
+        self.k = nn.Linear(c, c, bias=False)
+        self.v = nn.Linear(c, c, bias=False)
+        self.ma = nn.MultiheadAttention(embed_dim=c, num_heads=num_heads)
+        self.fc1 = nn.Linear(c, c, bias=False)
+        self.fc2 = nn.Linear(c, c, bias=False)
+        # Add layer normalization for stability
+        self.norm1 = nn.LayerNorm(c)
+        self.norm2 = nn.LayerNorm(c)
+        self.act = nn.GELU()  # Commonly used in transformers
```
Comment (AnhelinaRusak) [inline yolox/models/network_blocks.py:432]: 🟢Did you consider adding a dropout? From what I’ve read, it can help distribute attention more evenly and prevent overfitting

Comment (Pawel842) [reply yolox/models/network_blocks.py:432]: Yes, it's good idea, I have tried it, but the results were very similar so, I didn't added it.

**Review by AnhelinaRusak — APPROVED** [review]

**Review by msypniewskidtiq — APPROVED** [review]
